src/package_manager/CMakeLists.txt

```diff
@@ -0,0 +1,31 @@
+set(SOURCES packagemanagerfactory.cc
```

Comment:
What about this style?

```
set(SOURCES packagemanagerfactory.cc
            packagemanagerfake.cc
)

set(HEADERS packagemanagerfactory.h
            packagemanagerfake.h
            packagemanagerinterface.h)

aktualizr_source_file_checks(${SOURCES} ${HEADERS})
add_library(package_manager OBJECT ${SOURCES})

# debian backend
if(BUILD_DEB)
    target_sources(package_manager PUBLIC debianmanager.cc)
endif(BUILD_DEB)
aktualizr_source_file_checks(debianmanager.cc debianmanager.h)

# ostree backend
if(BUILD_OSTREE)
    target_sources(package_manager PUBLIC ostreemanager.cc ostreereposync.cc)
    target_include_directories(package_manager PUBLIC ${LIBOSTREE_INCLUDE_DIRS})
endif(BUILD_OSTREE)
aktualizr_source_file_checks(ostreemanager.cc ostreereposync.cc
                             ostreemanager.h ostreereposync.h)
```

Reply:
Nice idea. Will push an update to this one and src/utilities/CMakeLists.txt too

Reply:
Yes I think it streamlines the CMakeLists a little bit. After a few tries, it seems that `target_sources(... PRIVATE ...)` works better than `PUBLIC` actually.

`PUBLIC` modifies `INTERFACE_SOURCES` which we don't use and it caused some strange error when I tried to do the same in `src/aktualizr_secondary`. I'll push the changes on top of the PR.